mypyc/genops.py

```diff
@@ -800,8 +824,14 @@ def visit_name_expr(self, expr: NameExpr) -> Value:
         if isinstance(expr.node, Var):
             return self.environment.lookup(expr.node)
         if isinstance(expr.node, FuncDef):
-            # If we have a function, then we can look it up in the global variables dictionary.
-            return self.load_global(expr)
+            if expr.kind == LDEF:
+                namespace = self.get_function_namespace(expr.name)
+                if namespace:
+                    return self.load_function_class(expr, namespace)
```

Comment:
This still allocates a new object every time the nested function is referenced, which is needlessly inefficient. To be explicit, upon encountering a `def bar(): ...` inside the function `foo`, we should emit the following IR:
```
r0 = foo_bar_obj()
bar = r0
```
where `bar` is a Register created with `add_local`.

Then any references to `bar` should just use the `bar` register.

Reply:
Done.